### PR TITLE
Adds promo

### PR DESCRIPTION
**Pull request by NejcZdovc**
## Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).
- [x] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review). (Ask a Brave employee to help if you cannot access this document.)

## Test Plan:
- create folder `brave-development` and add `promoCode` file
- start browser and make sure that new tab with promo page is opened
- visit partner page and make sure that partner header is in the request via dev tools

## Reviewer Checklist:

- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review) if one was not already requested.

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

**Comment by bsclifton** [comment]
master f8c2de5197ec6823e49dcb88069c6397127ced09
0.23.x 0a7a128c39ea5ceaecb4c95d3779417900f4062c
0.22.x e5dd36a72bdc7f560429b52433acafdc3709b70c
0.22.x-promo 08b54da355f600995e0df82e42c2451b83395cb9

File: app/siteHacks.js [hunk 1]
```diff
@@ -8,78 +8,148 @@ const urlParse = require('./common/urlParse')
...
+      resourceName
+    }
   }
 
-  Filtering.registerBeforeSendHeadersFilteringCB((details, isPrivate) => {
-    const mainFrameUrl = Filtering.getMainFrameUrl(details)
-    if (!mainFrameUrl) {
-      return {
-        resourceName
+  const domain = urlParse(mainFrameUrl).hostname.split('.').slice(-2).join('.')
```
Comment (diracdeltas) [inline app/siteHacks.js]: this doesn't work for domains that have subdomains, ex: `play.spotify.com`

Comment (diracdeltas) [reply app/siteHacks.js]: you can use `getBaseDomain` to get the full domain

Comment (diracdeltas) [reply app/siteHacks.js]: though it looks like it also does this on master

Comment (NejcZdovc) [reply app/siteHacks.js]: yes this code is on the master as well, where I only moved this code into named funtion

**Pull request merged**